pyccel/codegen/printing/fcode.py

```diff
@@ -2219,8 +2227,17 @@ def _print_NumpyUfuncBase(self, expr):
 
     def _print_MathFunctionBase(self, expr):
         type_name = type(expr).__name__
+        if type_name == "MathPow":
+            code = self._print_Pow(expr)
```

Comment:
Please use `PyccelPow`

Reply:
I try to use this before
```python
if type_name == "MathPow":
            code = self._print_pyccelPow(expr)
```
but it gives this error `AttributeError: 'FCodePrinter' object has no attribute '_printPyccelPow'`

Reply:
ah solved. it was just a typo

Reply:
Remember you should rarely call `_print_XX` functions directly and they should always be passed an instance of the class indicated in the function name so you should be calling
`self._print(PyccelPow(expr))`
Not
`self._print_PyccelPow(expr)`
This should also make typos easier to catch as you know if the class exists even if you don't know if the `_print` function is implemented

Reply:
done

Reply:
it doesn't work.
```python
self = <pyccel.codegen.printing.fcode.FCodePrinter object at 0x7effdc629a60>
expr = PyccelPow(MathPow(x, y))

    def _print_PyccelPow(self, expr):
        base = expr.args[0]
>       e    = expr.args[1]
E       IndexError: tuple index out of range

../../pyccel/codegen/printing/fcode.py:2063: IndexError

```

Reply:
Ah ok sorry you want to use the `PyccelPow` printer to print a `MathPow`? As long as `PyccelPow` and `MathPow` have the same interface then that should be possible. In which case you are right you should call `self._print_PyccelPow(expr)`